lib/makeCmdTasks.js

```diff
@@ -31,20 +32,44 @@ export const makeCmdTasks = async ({ commands, cwd, files, gitDir, shell, verbos
     const resolvedArray = Array.isArray(resolved) ? resolved : [resolved] // Wrap non-array command as array
 
     for (const command of resolvedArray) {
-      // If the function linter didn't return string | string[]  it won't work
-      // Do the validation here instead of `validateConfig` to skip evaluating the function multiple times
-      if (isFn && typeof command !== 'string') {
-        throw new Error(
-          configurationError(
-            '[Function]',
-            'Function task should return a string or an array of strings',
-            resolved
+      if (isFn && command && typeof command === 'object') {
+        if (
+          typeof command.title === 'string' &&
+          typeof command.command === 'string' &&
+          typeof command.task === 'function'
+        ) {
+          const task = async (ctx = getInitialState()) => {
+            try {
+              await command.task()
+            } catch (e) {
+              throw makeErr(command.title, e, ctx)
+            }
+          }
+          cmdTasks.push({
+            title: command.title,
+            command: command.command,
+            task,
+          })
+        } else {
+          throw new Error(
+            configurationError('[Function]', 'Function task should be in proper format', resolved)
```

Comment:
can we provide a meaningful error message on what specifically was the problem?

Reply:
done